dora/src/bytecode/generator.rs

```diff
@@ -934,6 +935,25 @@ impl<'a> AstBytecodeGen<'a> {
             return self.visit_expr_dot_struct(expr, struct_id, type_params, dest);
         }
 
+        let call_type = self.src.map_calls.get(expr.id);
+        if call_type.map(|c| c.is_method()).contains(&true) {
+            let dot = ExprDotType {
+                id: expr.id,
+                pos: expr.pos,
+                span: expr.span,
+                lhs: expr.lhs.clone(),
+                rhs: expr.rhs.clone(),
+            };
+            let call = ExprCallType {
+                id: expr.id,
+                pos: expr.pos,
+                span: expr.span,
+                callee: Box::new(Expr::Dot(dot)),
+                args: Vec::new(),
+            };
+            return self.visit_expr_call(&call, dest);
+        }
+
```

Comment:
This is rather questionable, but way shorter than the code before that did everything manually.